lib/routes/security_pin/remote_server_auth/remote_server_auth.dart

```diff
@@ -0,0 +1,321 @@
+        }
+      }
+      _userController.text = data.user;
+      _passwordController.text = data.password;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final texts = context.texts();
+    final torBloc = AppBlocsProvider.of<TorBloc>(context);
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        leading: const backBtn.BackButton(),
+        title: Text(texts.remote_server_title),
+      ),
+      body: SingleChildScrollView(
+        reverse: true,
+        child: GestureDetector(
+          behavior: HitTestBehavior.opaque,
+          onTap: () => FocusScope.of(context).requestFocus(FocusNode()),
+          child: RemoteServerAuthForm(
+            formKey: _formKey,
+            urlController: _urlController,
+            failDiscoverURL: failDiscoverURL,
+            failNoBackupFound: failNoBackupFound,
+            userController: _userController,
+            failAuthenticate: failAuthenticate,
+            passwordController: _passwordController,
+          ),
+        ),
+      ),
+      bottomNavigationBar: Padding(
+        padding: const EdgeInsets.only(bottom: 0.0),
+        child: SingleButtonBottomBar(
+          stickToBottom: true,
+          text: widget.restore
+              ? texts.remote_server_action_restore
+              : texts.remote_server_action_save,
+          onPressed: () async {
+            final navigator = Navigator.of(context);
+
+            Uri uri = Uri.parse(_urlController.text);
+
+            bool connectionWarningResponse = true;
+            bool isAndroid = Platform.isAndroid;
+            if (isAndroid) {
+              if (uri.host.endsWith('onion') && torBloc.torConfig == null) {
+                await promptError(
+                  context,
+                  texts.remote_server_warning_connection_title,
+                  Text(
+                    texts.remote_server_warning_onion_message,
+                    style: Theme.of(context).dialogTheme.contentTextStyle,
+                  ),
+                  optionText: texts
+                      .remote_server_onion_warning_dialog_default_action_cancel,
+                  optionFunc: () {
+                    connectionWarningResponse = false;
+                    navigator.pop();
+                  },
+                  okText: texts.remote_server_onion_warning_dialog_settings,
+                  okFunc: () async {
+                    connectionWarningResponse = false;
+                    navigator.push(
+                      FadeInRoute(
+                        builder: (_) => withBreezTheme(
+                          context,
+                          const NetworkPage(),
+                        ),
+                      ),
+                    );
+                    return false;
+                  },
+                );
+              }
+            }
+            if (!uri.host.endsWith('.onion') && uri.scheme == 'http') {
+              promptAreYouSure(
```

Comment:
Tested with remote server over http and it breaks. Seems like it attempts to call testAuth before we allow connection.

Reply:
Resolved. 
- Refactor RemoteServerAuthPage auth logic 1e4fb77